crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -96,14 +98,27 @@ impl Solution {
                             buy: solution.prices
                                 [&fulfillment.order().buy.token.wrap(solution.weth)],
                         };
-                        let fulfillment = fulfillment.with_protocol_fees(prices)?;
-                        trades.push(Trade::Fulfillment(fulfillment))
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade)
                     }
                     order::Kind::Liquidity => {
                         trades.push(trade);
                     }
                 },
-                Trade::Jit(_) => trades.push(trade),
+                Trade::Jit(jit) => {
+                    if cow_amm_addresses.contains(&jit.order().receiver.into()) {
+                        let prices = ClearingPrices {
+                            sell: solution.prices[&jit.order().sell.token.wrap(solution.weth)],
+                            buy: solution.prices[&jit.order().buy.token.wrap(solution.weth)],
+                        };
+                        let trade = trade.with_protocol_fees(prices)?;
```

Comment:
No need to have protocol fees here. If you remember we jumped through a bunch of hoops to make CoW AMM orders exempt from protocol fees in the autopilot, remember?
Would be odd if we'd suddenly want to have fees when the trade is proposed as a JIT order. This PR should only be about computing the surplus for CoW AMM JIT orders and counting it towards the score of the solution.